vendor/config.py

```diff
@@ -1,7 +1,88 @@
 # App Settings
 from django.conf import settings
+from django import forms
+from django.db.models import TextChoices
+from django.contrib.sites.models import Site
+from django.core.exceptions import ObjectDoesNotExist
 from django.utils.translation import ugettext_lazy as _
 
+from siteconfigs.config import SiteConfigBaseClass
+from siteconfigs.models import SiteConfigModel
+
+class SupportedPaymentProcessor(TextChoices):
+    PROMO_CODE_BASE = ("base.PaymentProcessorBase", _("Default Processor"))
+    AUTHORIZE_NET = ("authorizenet.AuthorizeNetProcessor", _("Authorize.Net"))
+
+
+class PaymentProcessorForm(forms.Form):
+    processor = forms.CharField(label=_("Payment Processor"), widget=forms.Select(choices=SupportedPaymentProcessor.choices))
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.fields['processor'].widget = forms.Select(choices=SupportedPaymentProcessor.choices)
+        self.fields['processor'].label = _("Payment Processor")
+
+
+class PaymentProcessorSiteSelectForm(PaymentProcessorForm):
+    site = forms.CharField(label=_("Site"), widget=forms.Select(choices=[(site.pk, site.domain) for site in Site.objects.all()]))
+
+
+class PaymentProcessorSiteConfig(SiteConfigBaseClass):
+    label = _("Promo Code Processor")
+    default = {"payment_processor": "base.PaymentProcessorBase"}
+    form_class = PaymentProcessorForm
+    key = ""
+    instance = None
+
+    def __init__(self, site=None):
+        if site is None:
+            site = Site.objects.get_current()
+        self.key = ".".join([__name__, __class__.__name__])
+        self.set_instance(site)
+        super().__init__(site, self.key)
+
+    # TODO: This should be implemented in the SiteConfigBaseClass  
+    def save(self, valid_form):
+        site_config, created = SiteConfigModel.objects.get_or_create(site=self.site, key=self.key)
+        site_config.value = {"payment_processor": valid_form.cleaned_data['processor']}
+        site_config.site = self.site
```

Comment:
i don't think this site declaration is needed, since it's already taken care of by the get_or_create

Reply:
removed